src/components/order-service/order-service.controller.ts

```diff
@@ -312,6 +313,34 @@ export class OrderServiceController implements angular.IController {
     this.ctrl.parameterSchema = schema;
   }
 
+  private updateParameterForm() {
+    let formDefinition: any = _.get(this, 'ctrl.selectedPlan.externalMetadata.schemas.service_instance.create.form_definition');
+    this.removeUnsupportedParameterFormItems(formDefinition);
+    this.ctrl.parameterForm = formDefinition || ['*'];
+  }
+
+  private removeUnsupportedParameterFormItems(formItems: any) {
+    const allowedKeys = {
+      'key': true,
+      'type': true,
+      'items': true
+    };
+
+    _.forEach(formItems, (item: any) => {
+      if (_.isObject(item)) {
+        _.forOwn(item, (value, key) => {
+          if (!allowedKeys[key]) {
+            delete item[key];
```

Comment:
We should avoid mutating the original API object since it's cached by our DataService and possibly used outside of this controller. We should only modify a copy.